### DISCUSSION TITLE
How to access map values using DSL in Nuclei templates

### DISCUSSION THREAD

**Discussion by rushter** [projectdiscovery/nuclei#5704] [Q&A]
When checking for SSL protocol, there is a map of values:
```
[DBG] Ssl Protocol response variables:
        1. Host => xx.xx.xx.xx
        2. Hostname => xx.xx.xx.xx:443
        3. Port => 443
        4. cipher => TLS_AES_128_GCM_SHA256
        5. fingerprint_hash => {f4d67eaaa2068e5 .... c68494}
  ```

Is it possible to access each value? It looks like it's interpreted as a string in the DSL language.

Although, I can see that `tlsx` library outputs hashes as a map in the readme:
```
  "fingerprint_hash": {
    "md5": "c5208a47259d540a6e3404dddb85af91",
    "sha1": "df81dfa6b61eafdffffe1a250240db5d2e6cee25",
    "sha256": "7f2fe8d6b18e9a47839256cd97938daa70e8515750298ddba2f3f4b8440113fc"
  }
```

Right now, I'm using:
```
          - "contains(fingerprint_hash, 'HASH_VALUE')"
```

Is there a better way? I only want to compare sha256 hash.

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
